lib/ansible/modules/packaging/os/yum.py

```diff
@@ -1152,6 +1152,18 @@ def remove(self, items, repoq):
                     self.module.fail_json(**res)
 
             res['changed'] = True
+            if self.autoremove and not pkgs:
+                # bare autoremove invocation
+                candidates = (
+                    # CentOS 7.0-7.6
+                    'No Packages marked for removal',
+                    # Fedora 26-31
```

Comment:
The `yum` module in Ansible is also an [action plugin](https://docs.ansible.com/ansible/latest/plugins/action.html)[[0]](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/yum.py), which then in turn passes off to the appropriate backend package manager module, [yum](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/yum.py) or [dnf](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/dnf.py). Therefore, this candidate message should never be reached on Fedora, because the Ansible `dnf` module takes effect when the [dnf](https://dnf.readthedocs.io/en/latest/) package manager backend discovered (this is true on Fedora as well as RHEL8).

Reply:
With the `use_backend` option set to `yum` it should be still reachable on Fedora.

Reply:
@das7pad I don't think that it should. The only way to make that work is to install the `yum` package (RPM) and call out to the `yum-deprecated` command, which the Ansible module does not provide any means to accomplish so you're going to hit `dnf` either way and if you're attempting to use `use_backend: yum` on Fedora with the `yum` package (RPM) then you're going to find yourself querying system state with the YUM3 python APIs and then attempting to make changes via the `dnf` cli that's thinly masked as `/usr/bin/yum` and this will produce inconsistent and potentially undesired results because they don't share transaction/history databases. It possible I'm missing something, if so I'm happy to become informed and alter my stance on this.

Reply:
Ah, I see, there are python deps missing for yum on fedora.

Is this the case on RHEL 8 as well?

_Previously I just fired up docker containers for fedora 26 to 31, invoked `yum autoremove` and noted the output. Today I went for a full installation of fedora, and I can indeed verify the missing dependency._

Reply:
@das7pad yes, this is the case for RHEL8 as well. There is no longer yum 3.x in the context of RHEL8 so if you `use_backend: yum` on RHEL8 it will result in a python traceback.